cmd/chisel/cmd_find.go

```diff
@@ -0,0 +1,115 @@
+package main
+
+import (
+	"fmt"
+	"sort"
+	"strings"
+	"text/tabwriter"
+
+	"github.com/jessevdk/go-flags"
+
+	"github.com/canonical/chisel/internal/setup"
+	"github.com/canonical/chisel/internal/strdist"
+)
+
+var shortFindHelp = "Find existing slices"
+var longFindHelp = `
+The find command queries the chisel releases for available slices.
+
+With the --release flag, it queries for slices in a particular branch
+of the chisel-releases repository[1] or a particular directory. If left
+unspecified, it queries with the release info found in /etc/lsb-release.
+
+[1] https://github.com/canonical/chisel-releases
+`
+
+var findDescs = map[string]string{
+	"release": "Chisel release branch or directory",
+}
+
+type cmdFind struct {
+	Release string `long:"release" value-name:"<branch|dir>"`
+
+	Positional struct {
+		Query []string `positional-arg-name:"<query>" required:"yes"`
+	} `positional-args:"yes"`
+}
+
+func init() {
+	addCommand("find", shortFindHelp, longFindHelp, func() flags.Commander { return &cmdFind{} }, findDescs, nil)
+}
+
+func (cmd *cmdFind) Execute(args []string) error {
+	if len(args) > 0 {
+		return ErrExtraArgs
+	}
+
+	query := strings.TrimSpace(strings.Join(cmd.Positional.Query, " "))
+	if query == "" {
+		return fmt.Errorf("no search term specified")
+	}
+
+	release, releaseLabel, err := getRelease(cmd.Release)
+	if err != nil {
+		return err
+	}
+
+	slices, err := findSlices(release, query)
+	if err != nil {
+		return err
+	}
+	if slices == nil {
+		fmt.Fprintf(Stdout, "No matching slices for \"%s\"\n", query)
+		return nil
+	}
+
+	w := tabWriter()
+	fmt.Fprintf(w, "Slice\tPackage\tRelease\n")
+	for _, s := range slices {
+		fmt.Fprintf(w, "%s\t%s\t%s\n", s, s.Package, releaseLabel)
+	}
+	w.Flush()
+
+	return nil
+}
+
+const maxStrDist int64 = 1
```

Comment:
Given this is only used by `matchSlice`, could it be define as a const local to the function, and just called `maxDist` then?

Also it could be an untyped constant for simplicity.

Reply:
Nice, thanks. Changed in https://github.com/canonical/chisel/pull/99/commits/6d04d2d29fe6b1f7c3d1faa9c7bcb13a494877b8.